addons/mail/controllers/discuss.py

```diff
@@ -190,12 +190,12 @@ def fetch_image(self, channel_id, attachment_id, width=0, height=0, **kwargs):
     # --------------------------------------------------------------------------
 
     @http.route('/mail/init_messaging', methods=['POST'], type='json', auth='public')
-    def mail_init_messaging(self, **kwargs):
+    def mail_init_messaging(self, is_public_discuss=False, **kwargs):
         if not request.env.user.sudo()._is_public():
-            return request.env.user.sudo(False)._init_messaging()
+            return request.env.user.sudo(False).with_context(is_public_discuss=is_public_discuss)._init_messaging()
```

Comment:
Why not pass the context by rpc directly?

Reply:
I couldn't get it to work; maybe it doesn't work for routes?